engines/glk/scott/scott.cpp

```diff
@@ -385,13 +398,18 @@ void Scott::output(const Common::String &a) {
 		display(_bottomWindow, "%s", a.c_str());
 }
 
+void Scott::output(const Common::U32String &a) {
+	if (_saveSlot == -1)
+		display(_bottomWindow, Common::U32String("%S"), a.c_str());
+}
+
 void Scott::outputNumber(int a) {
 	display(_bottomWindow, "%d", a);
 }
 
 void Scott::look(void) {
-	const char *const ExitNames[6] = {
-		_("North").encode().c_str(), _("South").encode().c_str(), _("East").encode().c_str(), _("West").encode().c_str(), _("Up").encode().c_str(), _("Down").encode().c_str()
+	const uint32 *const ExitNames[6] = {
+		_("North").c_str(), _("South").c_str(), _("East").c_str(), _("West").c_str(), _("Up").c_str(), _("Down").c_str()
```

Comment:
You have a memory issue here.
TranslationManager::getTranslation() used to return a `const char *` pointer to a data in its translation messages array. But now it returns a `U32String` by copy. And here since you don't keep this `U32String` after getting the pointer to its data, it is destroyed. And you are thus keeping pointers to freed memory in the `ExitNames` array.

This is a similar issue to the one I highlighted in gtk-dialogs.cpp where `U32String::encode()` returns a `String` by copy and you are keeping a pointer to its data without keeping the `String` itself.

Reply:
Ah yes, makes sense. I will change it accordingly in a little while. Also, regarding your new commits, I will sweep the backends directory and search for `_(...)` usage, so I can catch any more compilation errors.